pyglet/shapes.py

```diff
@@ -258,6 +259,19 @@ def _update_vertices(self):
         """
         raise NotImplementedError("_update_vertices must be defined"
                                   "for every ShapeBase subclass")
+    @property
+    def rotation(self) -> float:
+        """Clockwise rotation of the shape in degrees.
+
+        It will be rotated about its (anchor_x, anchor_y)
+        position.
```

Comment:
It would probably be a good idea to add something like:   "which defaults to the first vertex point of the shape"

Reply:
I didn't add any notes on this because I figured we would be changing the rotation behavior soon.

How about something like the RST equivalent of this?

Shapes currently have inconsistent rotation behavior:

* Shapes default to rotating around their lower left corner
* Star, circle, ellipse, sector, and arc rotate around their centers

Reply:
> Shapes currently have inconsistent rotation behavior

It was discussed here but closed...

#840 & #841

Reply:
It's inconsistent, but kind of expected when dealing with OpenGL primitives. In any case we shouldn't break behavior for a 2.0 release. 

We can consider this change for 2.1 (development branch), if someone is willing to put in the work.
At a glance, I think the following would be a reasonable approach:
1. Add code to calculate the centroid of each Shape (might be tricky for some shapes), to determine the anchor offsets.
2. change `position` in the shader to a vec4, and place the anchor offset values there.

This should be in it's own thread for future discussion, if someone is willing to put the work in.

Reply:
I've done the following to address this thread's concerns:

1. Added a commit with the suggested information + a list of exceptions
2. Linked Ben's comment above in the relevant Discord thread so it isn't forgotten.